src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -1212,7 +1214,8 @@ private static DkmClrCompilationResultFlags GetLocalResultFlags(LocalSymbol loca
 
             foreach (var parameter in method.Parameters)
             {
-                if (GeneratedNames.GetKind(parameter.Name) == GeneratedNameKind.TransparentIdentifier)
+                if (GeneratedNames.GetKind(parameter.Name) == GeneratedNameKind.TransparentIdentifier ||
```

Comment:
N.B. The current closure conversion code for local functions may end up adding more struct closure environments than necessary to the closure method as parameters. Meaning that the closure may get environments containing variables they don't use. This is because of some implementation complexity in closure conversion that caused bookkeeping info for used/unused captured variables to get dropped before the time the struct closure environments are being gathered, so the capture code is conservative in selecting environments to capture.

The EE could try to find out the captured variables used in the given method and filter out unused struct environments, but I'm not sure that's worth the implementation complexity. I doubt anyone will be sad that they can view more locals than they thought.